components/brave_wallet/browser/internal/polkadot_extrinsic.rs

```diff
@@ -245,3 +270,156 @@ fn decode_unsigned_transfer_allow_death(
     let r = decode_unsigned_transfer_allow_death_impl(chain_metadata, input);
     Box::new(CxxPolkadotDecodeUnsignedTransferResult(r))
 }
+
+// Mortality tells the blockchain what range of blocks an extrinsic is valid
+// for. Validators use the period and phase to find the block hash included in
+// the payload, which they use to determine signature validity (and thus
+// transaction validity).
+//
+// We use a period of 64, as send thransactions should be of "high priority" and
+// thus subject to smaller mortality windows, i.e. we don't want an extrinsic
+// sitting in the memory pool while the world outside it continues to change.
+//
+// For information on how mortality is encoded, see the documentation at:
+// https://spec.polkadot.network/id-extrinsics#id-mortality
+//
+// Reference implementation:
+// https://github.com/polkadot-js/api/blob/9f6a9c53e6822d20e8556649c9b68d31cffc465d/packages/types/src/extrinsic/ExtrinsicEra.ts#L179-L204
+fn scale_encode_mortality(number: u32, period: u32) -> [u8; 2] {
+    let phase = number % period;
+    let factor = (period >> 12).max(1);
+
+    let left = 15.min((period.trailing_zeros() - 1).max(1));
+    let right = (phase / factor) << 4;
+    let encoded = u16::try_from(left | right).unwrap();
```

Comment:
I am not sure why in the reference code + is used instead of |, while in specs | should be used